src/homework/api/user.py

```diff
@@ -1,3 +1,29 @@
-from fastapi import APIRouter
+from typing import Annotated
+from fastapi import APIRouter, Depends, HTTPException
+from .contracts import UserBalanceResponse, UserBalanceRequest, AccessLevel
+from .security import ValidateHeader
+import src.homework.db.crud as crud
 
-router = APIRouter()
+
+router = APIRouter(
+    dependencies=[Depends(ValidateHeader(AccessLevel("Can_Read")))],
+)
+
+
+@router.get("/user")
+async def get_balance(
+    user_id: Annotated[UserBalanceRequest, Depends()]

```

Comment:
Could you elaborate, why do you use, 
```python
user_id: Annotated[UserBalanceRequest, Depends()]
```
instead of 
```python
user_id: UserBalanceRequest
```
Thanks in advance!

Reply:
I'm using the annotation this way according to [fastapi dependecies guide](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/) (I am using the [shortcut](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/#shortcut)). `Depends()` is necessary, because we use the class (pydantic model) as a type hint for the URL parameter. It will be validated by FastApi, and, in this case, it is equivalent to using `user_id: int`.

Reply:
Thanks for the explanation, I did not know about this shortcut!